src/service/video-manager-impl.js

```diff
@@ -192,57 +192,72 @@ class VideoEntry {
    * @private
    */
   loadedVideoVisibilityChanged_() {
-    if (this.canAutoplay_) {
+    if (this.hasAutoplay_) {
       this.autoplayLoadedVideoVisibilityChanged_();
     }
   }
 
-  /**
-   * Whether there is a need to monitor visibility of this video.
-   * @return {boolean}
-   * @package
-   */
-  needsVisibilityObserver() {
-    return this.canAutoplay_;
-  }
-
   /* Autoplay Behaviour */
 
   /**
    * Called when an autoplay video is built.
    * @private
    */
   autoplayVideoBuilt_() {
-    this.video.mute();
 
-    // If autoplay video has controls, hide them and only show them on
-    // user-ineraction.
-    if (this.video.element.hasAttribute(VideoAttributes.CONTROLS)) {
-      this.video.hideControls();
+    // Hide controls until we know if autoplay is supported, otherwise hiding
+    // and showing the controls quickly becomes a bad user experience for the
+    // common case where autoplay is supported.
+    this.video.hideControls();
 
-      // TODO(aghassemi): This won't work for iframes, needs a transparent shim
-      listenOnce(this.video.element, 'click', () => {
-        this.userInteracted_ = true;
+    this.boundSupportsAutoplay_().then(supportsAutoplay => {
+      if (!supportsAutoplay) {
+        // Autoplay is not supported, show the controls so user can manually
+        // initiate playback.
         this.video.showControls();
-        this.video.unmute();
-      });
-    }
+        return;
+      }
+
+      // Only muted videos are allowed to autoplay
+      this.video.mute();
+
+      // If autoplay video has controls, hide them and only show them on
+      // user-ineraction.
```

Comment:
Cool. Thanks for explanation. I think it might be worth exploring a similar UX for desktop as the mobile where controls timeout and are hidden after awhile in the future.